gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/UserServiceImpl.java

```diff
@@ -815,7 +822,11 @@ private UserEntity createAndSendEmail(
         }
 
         if (isBlank(newExternalUserEntity.getSourceId())) {
-            newExternalUserEntity.setSourceId(newExternalUserEntity.getEmail());
+            if (isServiceUser) {
+                newExternalUserEntity.setSourceId(newExternalUserEntity.getLastname());
+            } else {
+                newExternalUserEntity.setSourceId(newExternalUserEntity.getEmail());
+            }
```

Comment:
To avoid to many nested `if`:
```suggestion
            newExternalUserEntity.setSourceId(isServiceUser ? newExternalUserEntity.getLastname() : newExternalUserEntity.getEmail());
```